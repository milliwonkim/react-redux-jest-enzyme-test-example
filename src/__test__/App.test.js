import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import App from '../containers/App';

configure({ adapter: new Adapter() });

const mockStore = configureMockStore();
const store = mockStore({});

describe('App', () => {
	it('renders without crashing given the required props', () => {
		const props = {
			isFetching: false,
			dispatch: jest.fn(),
			selectedSubreddit: 'reactjs',
			posts: []
		};

		const wrapper = shallow(
			<Provider store={store}>
				<App {...props} />
			</Provider>
		);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
