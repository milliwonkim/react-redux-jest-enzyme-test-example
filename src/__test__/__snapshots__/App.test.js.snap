// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing given the required props 1`] = `
<Connect(App)
  dispatch={[MockFunction]}
  isFetching={false}
  posts={Array []}
  selectedSubreddit="reactjs"
/>
`;
